scientific_library/tvb/simulator/simulator.py

```diff
@@ -386,16 +388,18 @@ def __call__(self, simulation_length=None, random_state=None):
         local_coupling = self._prepare_local_coupling()
         stimulus = self._prepare_stimulus()
         state = self.current_state
+        start_step = self.current_step + 1
+        node_coupling = self._loop_compute_node_coupling(start_step)
 
         # integration loop
         n_steps = int(math.ceil(self.simulation_length / self.integrator.dt))
-        for step in range(self.current_step + 1, self.current_step + n_steps + 1):
-            # needs implementing by hsitory + coupling?
-            node_coupling = self._loop_compute_node_coupling(step)
+        for step in range(start_step, start_step + n_steps):
+            # needs implementing by history + coupling?
             self._loop_update_stimulus(step, stimulus)
             state = self.integrator.scheme(state, self.model.dfun, node_coupling, local_coupling, stimulus)
             self._loop_update_history(step, n_reg, state)
-            output = self._loop_monitor_output(step, state)
+            node_coupling = self._loop_compute_node_coupling(step + 1)
```

Comment:
I think this modify cation is not correct because this will give the coupling for the next time step and not for the current time step.
What do you think?

Reply:
interesting to shuffle around this way, but should be right